modules/telemetry/include/hephaestus/telemetry/struclog.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <iomanip>
+#include <source_location>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace heph::telemetry {
+
+static constexpr std::string_view LOG_PATTERN{ "level=%^%l%$ time=%Y-%m-%dT%H:%M:%S%z %v" };
+
+template <typename T>
+concept NotQuotable = !std::convertible_to<T, std::string> && requires(std::ostream& os, T t) {
+  { os << t } -> std::same_as<std::ostream&>;
+};
+
+template <typename T>
+concept Quotable = std::convertible_to<T, std::string>;
+
+template <typename T>
+struct Field final {
+  std::string key;
+  T val;
+};
+
+/**
+ * @brief A class that allows easy composition of logs for structured logging.
+ *        Example(see also struclog.cpp):
+ *        namespace ht=heph::telemetry;
+ *        LOG(INFO) << ht::Log{"adding"} | "id"_f(12345) | "tag"_f("test");
+ *
+ *        logs
+ *        'level=info file=struclog.h:123 time=2023-12-3T8:52:02+0 message="adding" id=12345
+ * tag="test"'
+ *
+ *        Remark: We would like to use libfmt with quoted formatting as proposed in
+ *                https://github.com/fmtlib/fmt/issues/825 once its included (instead of sstream).
+ *
+ */
+class Log {
+public:
+  explicit Log(const std::string& msg, std::source_location location = std::source_location::current());
```

Comment:
QQ: would it make sense to support fmt::format for msg? meaning adding support for variadic templates?